library/Faktory/Ent/Tracking.hs

```diff
@@ -0,0 +1,95 @@
+-- | Support for the @TRACK@ command (Enterprise only)
+--
+-- <https://github.com/contribsys/faktory/wiki/Ent-Tracking>
+--
+module Faktory.Ent.Tracking
+  ( CustomTrack(..)
+  , trackPerform
+
+  , JobDetails(..)
+  , JobState(..)
+  , trackGet
+  , trackGetHush
+
+  , SetJobDetails(..)
+  , trackSet
+  ) where
+
+import Faktory.Prelude
+
+import Control.Error.Util (hush)
+import Data.Aeson
+import Data.Aeson.Casing
+import qualified Data.ByteString.Lazy.Char8 as BSL8
+import Data.Maybe (fromMaybe)
+import Data.Time (UTCTime)
+import Faktory.Client (commandJSON, commandOK)
+import Faktory.Job (JobId, JobOptions, custom, perform)
+import Faktory.JobState (JobState(..))
+import Faktory.Producer
+import GHC.Generics (Generic)
+import GHC.Stack (HasCallStack)
+
+newtype CustomTrack = CustomTrack
+  { track :: Int
+  }
+  deriving stock Generic
+  deriving anyclass ToJSON
+
+-- | 'perform', but adding @{ custom: { track: 1 } }@
+--
+-- Equivalent to:
+--
+-- @
+-- 'perform' ('custom' $ 'CustomTrack' 1)
+-- @
+--
+trackPerform
+  :: (HasCallStack, ToJSON arg) => JobOptions -> Producer -> arg -> IO JobId
+trackPerform options = perform (options <> custom (CustomTrack 1))
+
+data JobDetails = JobDetails
+  { jdJid :: JobId
+  , jdPercent :: Maybe Int
+  , jdDesc :: Maybe Text
+  , jdState :: JobState
+  , jdUpdatedAt :: Maybe UTCTime
+  }
+  deriving stock Generic
+
+instance FromJSON JobDetails where
+  parseJSON = genericParseJSON $ aesonPrefix snakeCase
+
+unknownJobDetails :: JobId -> JobDetails
+unknownJobDetails jid = JobDetails
+  { jdJid = jid
+  , jdPercent = Nothing
+  , jdDesc = Nothing
+  , jdState = JobStateUnknown
+  , jdUpdatedAt = Nothing
+  }
+
+data SetJobDetails = SetJobDetails
+  { sjdJid :: JobId
+  , sjdPercent :: Maybe Int
+  , sjdDesc :: Maybe Text
+  , sjdReserveUntil :: Maybe UTCTime
+  }
+  deriving stock Generic
+
+instance ToJSON SetJobDetails where
+  toJSON = genericToJSON $ aesonPrefix snakeCase
+  toEncoding = genericToEncoding $ aesonPrefix snakeCase
+
+trackGet :: Producer -> JobId -> IO (Either String (Maybe JobDetails))
+trackGet producer jid =
+  commandJSON (producerClient producer) "TRACK GET" [BSL8.pack jid]
+
+-- | 'trackGet' but mask any failures to 'JobStateUnknown'
+trackGetHush :: Producer -> JobId -> IO JobDetails
```

Comment:
Gotcha so `Either .. (Maybe a) -> a`. At first I was wondering if the term "hush" was misleading since I usually think of it as resulting in a `Maybe` but hoogle search reveals a few different kinds of usage
![image](https://user-images.githubusercontent.com/1709849/119711533-a7072080-be2d-11eb-9468-e6c336e34f54.png)

Reply:
Oh snap, I totally misused "hush" here, I agree. Do you have any suggestions?

Reply:
I can't think of a better name off the top of my head. TBH it's kind of close to that `streamHush` :shrug: